festim/initial_condition.py

```diff
@@ -0,0 +1,63 @@
+import ufl
+from dolfinx import fem
+
+
+class InitialCondition:
+    """
+    Initial condition class
+
+    Args:
+        volume (festim.VolumeSubdomain1D): the volume subdomain where the condition is applied
+        value (float, int, fem.Constant or callable): the value of the initial condition
+        species (festim.Species): the species to which the condition is applied
+
+    Attributes:
+        volume (festim.VolumeSubdomain1D): the volume subdomains where the condition is applied
+        value (float, int, fem.Constant or callable): the value of the initial condition
+        species (festim.Species): the species to which the source is applied
+
+    Usage:
+        >>> from festim import InitialCondition
+        >>> InitialCondition(volume=my_vol, value=1, species="H")
+        >>> InitialCondition(volume=my_vol, value=lambda x: 1 + x[0], species="H")
+        >>> InitialCondition(volume=my_vol, value=lambda t: 1 + t, species="H")
+        >>> InitialCondition(volume=my_vol, value=lambda T: 1 + T, species="H")
+        >>> InitialCondition(volume=my_vol, value=lambda x, t: 1 + x[0] + t, species="H")
```

Comment:
According to the docstrings, the `species` argument is of type `festim.Species` not `str`